base/src/main/java/com/mindera/skeletoid/routing/OpenUriCommand.kt

```diff
@@ -1,19 +1,26 @@
 package com.mindera.skeletoid.routing
 
+import android.content.ActivityNotFoundException
 import android.content.Context
 import android.content.Intent
 import android.net.Uri
+import com.mindera.skeletoid.logs.LOG
 
 class OpenUriCommand(private val context: Context, private val deepLink: String) : IRouteCommand {
 
     companion object {
+        private const val LOG_TAG = "OpenUriCommand"
         const val IS_INTERNAL_APP_NAVIGATION_EXTRA = "IS_INTERNAL_APP_NAVIGATION_EXTRA"
     }
 
     override fun navigate() {
         val intent = Intent(Intent.ACTION_VIEW)
         intent.data = Uri.parse(deepLink)
         intent.putExtra(IS_INTERNAL_APP_NAVIGATION_EXTRA, true)
-        context.startActivity(intent)
+        try {
+            context.startActivity(intent)
+        } catch (exception: ActivityNotFoundException) {
+            LOG.e(LOG_TAG, exception, "Could not find an activity to handle $intent")
+        }
```

Comment:
I'm not very fond of this solution. Maybe we can see if there's any application which can open that intent and only then decide whether to call start activity or throw exception. 

check this [link](https://developer.android.com/training/basics/intents/sending) — section **Verify There is an App to Receive the Intent** — for more info 

I still think we should throw an exception. This is a routing command which lives independently of an application so where to are we logging?

Reply:
I thought of that, but if we want to throw an exception then we can leave as it and let the app that is using this handle it itself, right? Since it already has a specific exception for it

The idea was to end the propagation there and just send the logs for it. But I get that no feedback at all is also a bad practice

Reply:
Should we let it as it was then? 🤔

Reply:
I was thinking that we could throw a custom exception. Either this or leave as it was.

You can also add an annotation to flag this can throw the exception.

Reply:
I'm okay with either one of them. Which one do you prefer?

Reply:
I guess this is the most explicit exception so leave it as it is?

Reply:
I guess this is the most explicit exception so leave it as it is?